### ISSUE TITLE
/en-US/search-index.json isn't working

### ISSUE THREAD

**Issue by peterbe** [mdn/yari#976]
Note, this is happening in the reductio-reloaded branch

Steps to reproduce:

1. `yarn start`
2. Go to http://localhost:3000/en-US/search-index.json
3. Click into (hover over) the search input and this happens:
<img width="754" alt="Screen Shot 2020-07-28 at 3 58 12 PM" src="https://user-images.githubusercontent.com/26739/88714856-5dee1f00-d0eb-11ea-9d55-c207de339cf3.png">

There are two problems here. First of all, it shouldn't say "Initializing index..." when it actually failed. Also, the placeholder still says "Go ahead. Type your search..."

The reason it's failing in the first place is that, in `yarn start` is because of this:
```
▶ curl -v http://localhost:5000/en-US/search-index.json
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 5000 (#0)
> GET /en-US/search-index.json HTTP/1.1
> Host: localhost:5000
> User-Agent: curl/7.64.1
> Accept: */*
>
< HTTP/1.1 200 OK
< X-Powered-By: Express
< Content-Type: application/json; charset=utf-8
< Date: Tue, 28 Jul 2020 20:01:47 GMT
< Connection: keep-alive
< Content-Length: 0
<
* Connection #0 to host localhost left intact
* Closing connection 0

```
which means it returned (immediately) with an empty body. 
We should instead simulate what would happen in production which is that it should simply wait until it's ready.

**Comment by peterbe** [comment]
Note-to-self; I've only tested this in https://github.com/Gregoor/yari/pull/9 's branch but I think it's up-to-date with reductio-reloaded.

**Comment by Gregoor** [comment]
Fun that I managed to break it for dev, without breaking the tests 🙈 Reason was that the search index couldn't handle the change to lowercased request URLs.  Now it can.

Given that watching takes ~30s with the current content already on some machines (like mine), I'd prefer showing a message there, even though it's code that won't be relevant for production. It does make for a nicer local mode experience imo.

I also made the change to not show the initializing thing when an error occurred.
